include/h2o/absprio.h

```diff
@@ -22,22 +22,36 @@
 #ifndef h2o__absprio_h
 #define h2o__absprio_h
 
-typedef enum en_h2o_absprio_urgency_type_t {
-    H2O_ABSPRIO_URGENCY_PREREQUISITE = 0,
-    H2O_ABSPRIO_URGENCY_DEFAULT = 1,
-    H2O_ABSPRIO_URGENCY_SUPPLEMENTARY_0 = 2,
-    H2O_ABSPRIO_URGENCY_SUPPLEMENTARY_1 = 3,
-    H2O_ABSPRIO_URGENCY_SUPPLEMENTARY_2 = 4,
-    H2O_ABSPRIO_URGENCY_SUPPLEMENTARY_3 = 5,
-    H2O_ABSPRIO_URGENCY_SUPPLEMENTARY_4 = 6,
-    H2O_ABSPRIO_URGENCY_BACKGROUND = 7,
-} h2o_absprio_urgency_type_t;
+#include <assert.h>
+#include <stddef.h>
+#include <stdint.h>
 
-void h2o_absprio_parse_priority(const h2o_iovec_t *header_value, uint8_t *urgency, int *incremental);
+#define H2O_ABSPRIO_URGENCY_PREREQUISITE 0
```

Comment:
Any particular reason for preferring macro over enum?

Reply:
I think the question boils down to if we consider urgency as a number or an enumeration.

I tend to think it as a number, of which some tend to have specific meanings. If we view it as numbers, it should be consistently referred to by using `uint8_t` rather than sometimes using `h2o_absprio_urgency_type_t`.

That is why I decided to change the enum to a list of constants. And the most straightforward way of defining constants that are distinct from each other (i.e. URGENCY_BACKGROUND is 7 by definition, rather than SUPPLEMENTARY + 1) is to use `#define`.